### PR TITLE
Keras callbacks support

### PR DESCRIPTION
**Pull request by rachaelvp**
- User can now supply a list of callback functions to keras learners. 
- Added one default callback, an early stopping criteria, which stops training when a monitored quantity (default is validation loss) has stopped improving after `patience` (default is 10) number of epochs.

### CODE REVIEW

**Comment by nhejazi** [comment]
@rachaelvp Even though this is now merged into `devel`, please create a new PR that bumps the `devel` version from 1.4.2 (that's what `master` is on) to 1.4.3 in the `DESCRIPTION` file, and also document the changes introduced in this PR appropriately in the [`NEWS.md` file](https://github.com/tlverse/sl3/blob/devel/NEWS.md), since we'll have to keep `NEWS.md` appropriate updated going forward.
